File: php/min-depth-binary-tree.php
<?php

class Node {
    public $data;
    public $left;
    public $right;

    public function __construct($data) {
        $this->data = $data;
        $this->left = null;
        $this->right = null;
    }
}

class Tree {
    public $root;
}

class Solution {
    public function minDepth(Node $root = null):  int {
        if (is_null($root)) {
            return 0;
        }

        if (is_null($root->left) && is_null($root->right)) {
            return 1;
        }

        if (!$root->left) {
            return $this->minDepth($root->right) + 1;
        }

        if (!$root->right) {
            return $this->minDepth($root->left) + 1;
        }

        return min($this->minDepth($root->left), $this->minDepth($root->right)) + 1;
    }
}

$root = new Node(1);
$root->left = new Node(9);
$root->right = new Node(20);
$root->right->left = new Node(15);
$root->right->right = new Node(7);

$solution = new Solution();
echo $solution->minDepth($root) . "\n"; // 2

File: python/min-depth-binary-tree.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class Tree:
    def __init__(self, root):
        self.root = root
class Solution:
    def minDepth(self,root):
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 1
        if root.left is None:
            return self.minDepth(root.right)+1
        if root.right is None:
            return self.minDepth(root.left)+1
        return min(self.minDepth(root.left), self.minDepth(root.right))+1
    
root = Node(1)
root.left = Node(9)
root.right = Node(20)
root.right.left = Node(15)
root.right.right = Node(7)
tree = Tree(root)
print(Solution().minDepth(tree.root))